intdataplane/int_dataplane.go

```diff
@@ -591,6 +606,31 @@ func (d *InternalDataplane) apply() {
 		}
 		d.cleanupPending = false
 	}
+
+	// Set up any needed rescheduling kick.  Any pending kick will have been recalculated above
+	// so we always cancel and reschedule.
+	if d.reschedC != nil {
+		// We have an active rescheduling timer, stop it so we can reset it below if needed.
+		// This snippet comes from the docs for Timer.Stop().
+		if !d.reschedTimer.Stop() {
+			// Timer had already popped, drain its channel.
+			<-d.reschedC
+		}
+		// Nil out our copy of the channel to record that the timer is inactive.
+		d.reschedC = nil
+	}
+	if reschedDelay != 0 {
+		// We need to reschedule.
+		log.WithField("delay", reschedDelay).Debug("Asked to reschedule.")
+		if d.reschedTimer == nil {
+			// First time, create the timer.
+			d.reschedTimer = time.NewTimer(reschedDelay)
```

Comment:
Is it possible to create a timer without actually starting it?  If so, I think it would be nicer to do this in the dataplane constructor, instead of JIT here.